src/transport_picoquic.cpp

```diff
@@ -738,73 +739,62 @@ PicoQuicTransport::close([[maybe_unused]] const TransportContextId& context_id)
 {
 }
 
-void
+bool
 PicoQuicTransport::checkTxData()
 {
+    bool available = false;
     // std::map<TransportContextId, std::map<StreamId, StreamContext>> active_streams;
     for (auto& [tcid, streams] : active_streams) {
         for (auto& [sid, stream] : streams) {
-
             if (sid != 0) {
-                // QUIC Stream
-                if (stream.tx_data->size() > 0)
-                    (void)picoquic_mark_active_stream(
-                      stream.cnx, sid, 1, static_cast<StreamContext*>(&stream));
+                if (stream.tx_data->size() > 0) {
+                    (void) picoquic_mark_active_stream(
+                            stream.cnx, sid, 1, static_cast<StreamContext *>(&stream));
+                }
             } else {
                 // QUIC Datagram
                 if (stream.tx_data->size() > 0) {
-                    // reinsert by wake is one way to get picoquic to callback sooner
-                    // picoquic_reinsert_by_wake_time(s_pair.second.cnx->quic, s_pair.second.cnx, cur_time);
-                    // Only send tx data here when using picoquic queueing, otherwise let the callbacks handle it
-                    // sendTxData(&s_pair.second, NULL, 1400);
-                    //picoquic_mark_datagram_ready(stream.cnx, 1);
-                    update_datagram_ready_status(stream.cnx, true);
-                } else {
-                    //picoquic_mark_datagram_ready(stream.cnx, 0);
-                    update_datagram_ready_status(stream.cnx, false);
+                    available = true;
+                    break;
                 }
             }
         }
     }
+
+    return available;
 }
 
 void
 PicoQuicTransport::sendTxData(StreamContext* stream_cnx, [[maybe_unused]] uint8_t* bytes_ctx, size_t max_len)
 {
     if (bytes_ctx == nullptr) {
+        //std::cerr << "N;";
         metrics.send_null_bytes_ctx++;
         return;
     }
 
+
     const auto& out_data = stream_cnx->tx_data->front();
     if (out_data.has_value()) {
         if (max_len >= out_data.value().size()) {
             metrics.dgram_sent++;
-
             uint8_t* buf = nullptr;
-
             if (stream_cnx->stream_id == 0) {
                 buf = picoquic_provide_datagram_buffer_ex(bytes_ctx,
                                                           out_data.value().size(),
-                                                          1);
+                                                          (stream_cnx->tx_data->size() - 1) > 0);
```

Comment:
I assume this is safe because size couldn't be 0 if we're in this `if` block, right?